Whitelist email support

ajvb opened the pull request:
Adds support for `/whitelist_email` (and `/staging_whitelist_email`) that functions exactly like `/whitelist_ip`.

Whitelisted objects are stored under the same namespace with different datastore kinds.

This PR includes an additional change for moving the checking of the pubsub topic existing in SlackbotHTTP into the init phase rather than on each http request.

Fixes #21 
Depends on https://bugzilla.mozilla.org/show_bug.cgi?id=1564951

ajvb committed:
Add support for whitelisting emails

Move topic exists check to InitConfig

Handle else case for whitelisted type switching

ameihm0912 commented on slackbot-background/handlers.go:
```diff
@@ -13,9 +13,21 @@ import (
+	}
+	if cmd.Cmd == WHITELIST_EMAIL_SLASH_COMMAND || cmd.Cmd == STAGING_WHITELIST_EMAIL_SLASH_COMMAND {
+		whitelistedObj.Type = common.EMAIL_TYPE
+	}
```
Does the above block need to be an `if..else if..else { return err }`?

ameihm0912 commented on slackbot-background/utils.go:
```diff
@@ -44,27 +53,48 @@ func checkUsersGroups(email string) (bool, error) {
+		} else {
+			obj = ip.String()
+		}
+	}
```
similar to previous comment, if the above blocks don't match `err` will be non-nil and we will proceed past the block below

ajvb commented:
@ameihm0912 Yeah totally. I originally didn't handle it because there is the `allowedCommand` (https://github.com/mozilla-services/foxsec-pipeline-contrib/pull/23/files#diff-4c827819529ac4ad14a2e4a91b963fd6R102) check before these functions are called, but not very defensive of me. Fixed and squashed the commits.

adrianosela commented on slackbot-background/utils.go:
```diff
@@ -98,12 +128,12 @@ func isAlertConfirm(callbackId string) bool {
+		log.Infof("Sending DELETE request to %s for %s/%s", iprepdInstance.URL, typestr, obj)
 
-		req, err := http.NewRequest("DELETE", iprepdInstance.URL+"/"+ip, nil)
+		req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/type/%s/%s", iprepdInstance.URL, typestr, obj), nil)
```
If you want to use the iprepd client lib:

```
c, err := iprepd.NewClient(iprepdInstance.URL, iprepdInstance.APIKey, client)
checkErr(err)
```
and in the loop:

```
log.Infof("Sending DELETE request to %s for %s/%s", iprepdInstance.URL, typestr, obj)
if err := c.DeleteReputation(typestr, obj); err != nil {
        log.Errorf("could not delete reputation entry for %s/%s: %s", typ, obj, err)
}
```

adrianosela reviewed (approved):
LGTM 👍 left one comment suggesting to use the iprepd client lib here
